pkg/controlplane/apiserver/server.go

```diff
@@ -230,6 +230,18 @@ func (c completedConfig) New(name string, delegationTarget genericapiserver.Dele
 				err := c.Extra.PeerProxy.WaitForCacheSync(context.Done())
 				return err
 			})
+
+			// Register the new merged discovery endpoints.
+			s.GenericAPIServer.AddPostStartHookOrDie("register-merged-discovery-endpoints", func(context genericapiserver.PostStartHookContext) error {
+				mergedHandler := c.Generic.MergedDiscoveryHandler
+				if mergedHandler == nil {
+					return fmt.Errorf("MergedDiscoveryHandler not configured")
+				}
+
+				s.GenericAPIServer.Handler.NonGoRestfulMux.HandleFunc("/api/merged", mergedHandler.ServeHTTP)
+				s.GenericAPIServer.Handler.NonGoRestfulMux.HandleFunc("/apis/merged", mergedHandler.ServeHTTP)
```

Comment:
Merged discovery isn't a new sub-resource, but rather a different representation of the existing /api and /apis resources.

My understanding is that the returned Kind would be the same in both cases. The key difference is that the existing resources return a single level of data, while a merged view would return all levels.

We have a precedent for requesting different representations with `Accept: application/json;as=Table;v=v1beta1;g=meta.k8s.io`. However, we can't use the as parameter here, since discovery would use `as=APIResourceDiscovery;v=v2beta1;g=apidiscovery.k8s.io` for both the normal and merged views.

What about using a different parameter, like `application/json;profile=Merged` or `application/json;scope=All`?

Reply:
Per offline discussion, I think we can returned merged discovery by default when the feature is on, and only serve unmerged discovery when peers specifically ask for it.  This way, discovery and the APIs actually made available by an apiserver to agree.

The fallback here is important.  For clients that do provide `profile=Unmerged` (or whatever we decide to call it), if the server doesn't support merged discovery, it returned unmerged data, and if the server *does* support merged discovery, it will respect the param and served unmerged data.  So a Accept header works great.